src/client.rs

```diff
@@ -555,9 +562,27 @@ async fn run_producer<Exe: Executor>(
     }) = messages.next().await
     {
         match producer.send(topic, payload).await {
-            Ok(future) => {
+            Ok(send_f) => {
+                let delay_f = client
+                    .executor
+                    .delay(client.operation_retry_options.operation_timeout);
+
                 let _ = client.executor.spawn(Box::pin(async move {
-                    let _ = resolver.send(future.await);
+                    pin_mut!(delay_f);
+                    pin_mut!(send_f);
```

Comment:
Is there a reason why send_f should also be pinned?

Reply:
My understanding is that since send_f: SendFuture has already been moved in line 570, when select returns with Either::Right(), the send_f will be dropped, thus pin_mut is unnecessary.